app/src/playStore/java/org/fossasia/openevent/general/search/LocationServiceImpl.kt

```diff
@@ -0,0 +1,72 @@
+package org.fossasia.openevent.general.search
+
+import android.content.Context
+import android.location.Address
+import android.location.Geocoder
+import android.location.LocationManager
+import com.google.android.gms.location.*
+import io.reactivex.Single
+import org.fossasia.openevent.general.utils.LocationPermissionException
+import org.fossasia.openevent.general.utils.NoLocationSourceException
+import timber.log.Timber
+import java.io.IOException
+import java.util.*
+
+class LocationServiceImpl(private val context: Context) : LocationService {
+
+    override fun getAdministrativeArea(): Single<String> {
+        val locationManager = context.getSystemService(Context.LOCATION_SERVICE)
+
+        if (locationManager !is LocationManager) {
+            return Single.error(IllegalStateException())
+        }
+
+        if (!locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER)) {
+            return Single.error(NoLocationSourceException())
+        }
+
+        val locationRequest = LocationRequest.create().apply {
+            priority = LocationRequest.PRIORITY_LOW_POWER
+        }
+        return Single.create { emitter ->
+            try {
+                LocationServices
+                    .getFusedLocationProviderClient(context)
+                    .requestLocationUpdates(locationRequest, object : LocationCallback() {
+
+                        override fun onLocationResult(locationResult: LocationResult?) {
+                            try {
+                                val adminArea = locationResult.getAdminArea()
+                                emitter.onSuccess(adminArea)
+                            } catch (e: IllegalArgumentException) {
+                                emitter.onError(e)
+                            }
+                        }
+                    }, null)
+            } catch (e: SecurityException) {
+                emitter.onError(LocationPermissionException())
+            }
+        }
+    }
+
+    private fun LocationResult?.getAdminArea(): String {
+        if (this == null) {
```

Comment:
this can never be null

Reply:
I just took it from the old code. I'll remove this